### PR TITLE
Update -fsycl-targets option values

### PR DESCRIPTION
**Pull request by bader**
`-sycldevice` triple suffix is not needed anymore.
This change should fix the warning emitted by the latest version of DPC++ compiler.

### CODE REVIEW

File: SYCL/Basic/image/image_read.h [hunk 1]
```diff
@@ -7,7 +7,7 @@
  *
  * https://www.khronos.org/registry/OpenCL/specs/opencl-1.2.pdf#page=329
  *
- * Confirmed to pass with -fsycl-targets=spir64-unknown-linux-sycldevice
+ * Confirmed to pass with -fsycl-targets=spir64-unknown-linux
```
Comment (bader) [inline SYCL/Basic/image/image_read.h]: I have no idea what this comment is about. Can we delete it?

File: SYCL/SpecConstants/2020/non_native/cuda.cpp [hunk 2]
```diff
@@ -1,6 +1,6 @@
 // REQUIRES: cuda
 
-// RUN: %clangxx -fsycl -fsycl-targets=nvptx64-unknown-unknown-sycldevice %S/Inputs/common.cpp -o %t.out
+// RUN: %clangxx -fsycl -fsycl-targets=nvptx64-nvidia-cuda %S/Inputs/common.cpp -o %t.out
 // RUN: env SYCL_DEVICE_FILTER=cuda %t.out
 
 // TODO: enable this test then compile-time error in sycl-post-link is fixed
```
Comment (bader) [inline SYCL/SpecConstants/2020/non_native/cuda.cpp:6]: I noticed that target triple is wrong. Can it be the reason of "compile-time error in sycl-post-link"?
Should I try to enable this test?

File: SYCL/Basic/image/image_read.h (hunk 1 above)
Comment (AGindinson) [reply SYCL/Basic/image/image_read.h]: Should be good to remove, looks quite outdated

File: SYCL/SpecConstants/2020/non_native/cuda.cpp (hunk 2 above)
Comment (AGindinson) [reply SYCL/SpecConstants/2020/non_native/cuda.cpp:6]: I can't evaluate if `sycl-post-link` would be the tool affected by the incorrect triple or not, but personally, wouldn't object against a CI check as to that.

**Review by AGindinson — COMMENTED** [review]
The changes as such LGTM

**Review by AGindinson — APPROVED** [review]

**Review by dm-vodopyanov — APPROVED** [review]

**Review by vladimirlaz — APPROVED** [review]

**Comment by bader** [comment]
@kbobrovs, @smaslov-intel, @vzakhari, @AlexeySachkov, @Fznamznon, could you take a look, please?

**Review by smaslov-intel — APPROVED** [review]

**Review by vzakhari — APPROVED** [review]

**Review by vladimirlaz — APPROVED** [review]

**Review by romanovvlad — APPROVED** [review]

**Pull request merged**